### ISSUE TITLE
Cannot enable cors using api gateway with --handler=api

### ISSUE THREAD

**Issue by phuonghau98** [micro/go-micro#894]
I have spent hours trying to enables cors (also other plugins)
I have a service and a gateway. I don't know is the readme is not updated or i didn't do it correctly
Firstly, I started my service by following: **go run main.go plugin.go**
Then, I started my gateway using: **micro api --handler=api**
Everytime i fetch data from react-app, i get: ** request has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.**
Thanks in advance.

**Comment by mirwaisx** [comment]
Hi Phuong
You are not alone i had similar issues and took a few days until i figured it out, among other go modules related problems.
It appears that those micro plugins and NOT go-micro plugins, need micro binary to be rebuilt with the plugins.go file.
https://github.com/micro/go-plugins/tree/master/micro
You can take a look in the readme.md file of headers lkugin for example. 

So you dont do go run main.go plugin.go but inside micro toolkit project rebuild it with your plugin.go file: make build should be enogh if plugin.go is in the same folder as makefile.
Now if this is the correct way of rebuilding micro always when you want to add a plugin, I am not sure, but it solved the plugin issue.

**Comment by allenkaplan** [comment]
Hey Phuong. I am also having some issues enabling CORS. I have been led to believe that you must run your api with `micro --plugin [plugin] api`. I am running into issues as I cannot enable the right plugin. When I attempt to use `go-micro/micro/cors` or just `micro/cors` I get ` | Error loading plugin go-micro/micro/cors: Failed to build plugin cors: exit status 1` and ` | Error loading plugin micro/cors: Failed to build plugin cors: exit status 1` respectively. When attempting to use simply `cors` I get `Error loading plugin cors: Unknown plugin cors`

If anyone can figure out the correct way of enabling the plugin it would be greatly appreciated. I have not found any information regarding needing to rebuild the micro binary and I believe that there is an issue with the go-micro plugin.

**Comment by mirwaisx** [comment]
>  I have not found any information regarding needing to rebuild the micro binary and I believe that there is an issue with the go-micro plugin.

see this: https://github.com/micro/go-plugins/tree/master/micro/cors#usage or this https://github.com/micro/go-plugins/tree/master/micro/header#usage: "rebuild micro with the plugin"

For me it's working fine: I do not use CORS but a custom plugin adding a 'Copntent-Type' header if it is missing. I had to do this as there is a bug #889  go-micro if the COntent-Type is missing

**Comment by allenkaplan** [comment]
Thanks for the response @mirwaisx. I have rebuilt the binary, but I am now running into another issue regarding using the CORS plugin: https://github.com/micro/go-plugins/issues/419 with the api handler

**Comment by duyndh** [comment]
I having the same issues. Despite of rebuilding the main file with plugin file. 
- Micro version: 1.15.1
